gtda/time_series/embedding.py

```diff
@@ -75,13 +79,19 @@ class SlidingWindow(BaseEstimator, TransformerResamplerMixin):
 
     _hyperparameters = {
         'width': {'type': int, 'in': Interval(1, np.inf, closed='left')},
-        'stride': {'type': int, 'in': Interval(1, np.inf, closed='left')}
+        'stride': {'type': int, 'in': Interval(1, np.inf, closed='left')},
+        'target_resampler': {'type': Callable}
     }
 
-    def __init__(self, width=10, stride=1):
+    def __init__(self, width=10, stride=1, target_resampler=None):
         self.width = width
         self.stride = stride
 
+        if target_resampler is None:
```

Comment:
Parameters logic should not be in the init of a sklearn-estimator. Move the logic in fit and use it to define an attribute as we do for ``metric_params`` in `diagrams.PairwiseDistances` for example